src/graphics/program-lib/standard.js

```diff
@@ -1,4 +1,98 @@
+_oldChunkWarn = function(oldName, newName) {
+    // #ifdef DEBUG
+    console.warn("Shader chunk " + oldName + " is deprecated - override " + newName + " instead");
+    // #endif
+}
+
+var _oldChunkFloat = function(s, o, p) {
+    _oldChunkWarn(p, o);
+    return "\n#ifdef MAPFLOAT\n" + s + "\n#else\n" + pc.shaderChunks[o] + "\n#endif\n";
+}
+
+var _oldChunkColor = function(s, o, p) {
+    _oldChunkWarn(p, o);
+    return "\n#ifdef MAPCOLOR\n" + s + "\n#else\n" + pc.shaderChunks[o] + "\n#endif\n";
+}
+
+var _oldChunkTex = function(s, o, p) {
+    _oldChunkWarn(p, o);
+    return "\n#ifdef MAPTEXTURE\n" + s + "\n#else\n" + pc.shaderChunks[o] + "\n#endif\n";
+}
+
+var _oldChunkTexColor = function(s, o, p) {
+    _oldChunkWarn(p, o);
+    return "#undef MAPTEXTURECOLOR\n#ifdef MAPTEXTURE\n#ifdef MAPCOLOR\n#define MAPTEXTURECOLOR\n#endif\n#endif\n" + 
+            "#ifdef MAPTEXTURECOLOR\n" + s + "\n#else\n" + pc.shaderChunks[o] + "\n#endif\n";
+}
+
+var _oldChunkTexFloat = function(s, o, p) {
+    _oldChunkWarn(p, o);
+    return "#undef MAPTEXTUREFLOAT\n#ifdef MAPTEXTURE\n#ifdef MAPFLOAT\n#define MAPTEXTUREFLOAT\n#endif\n#endif\n" + 
+            "#ifdef MAPTEXTUREFLOAT\n" + s + "\n#else\n" + pc.shaderChunks[o] + "\n#endif\n";
+}
+
+var _oldChunkVert = function(s, o, p) {
+    _oldChunkWarn(p, o);
+    return "\n#ifdef MAPVERTEX\n" + s + "\n#else\n" + pc.shaderChunks[o] + "\n#endif\n";
+}
+
+var _oldChunkVertColor = function(s, o, p) {
+    _oldChunkWarn(p, o);
+    return "#undef MAPVERTEXCOLOR\n#ifdef MAPVERTEX\n#ifdef MAPCOLOR\n#define MAPVERTEXCOLOR\n#endif\n#endif\n" + 
+            "#ifdef MAPVERTEXCOLOR\n" + s + "\n#else\n" + pc.shaderChunks[o] + "\n#endif\n";
+}
+
+var _oldChunkVertFloat = function(s, o, p) {
+    _oldChunkWarn(p, o);
+    return "#undef MAPVERTEXFLOAT\n#ifdef MAPVERTEX\n#ifdef MAPFLOAT\n#define MAPVERTEXFLOAT\n#endif\n#endif\n" + 
+            "#ifdef MAPVERTEXFLOAT\n" + s + "\n#else\n" + pc.shaderChunks[o] + "\n#endif\n";
+}
+
 pc.programlib.standard = {
+
+    _oldChunkToNew: {
+        aoTexPS : {n:"aoMapPS", f:_oldChunkTex},
```

Comment:
If we doing all this work to support old chunk names, can we get rid of "PS" as GLSL doesn't use pixel shaders. And also doesn't fit our coding standards for acronyms (should be `aoTexFs`?)

Reply:
Next time, in another PR ;)

Reply:
Then we're going to break any shader chunk code twice?

Reply:
This PR is made carefully to NOT break any shader chunk code.

Reply:
But doing setters for all chunks we have for the sake of changing names isn't useful

Reply:
So it doesn't break shader chunk code, but does break javascript?

Reply:
diffuseMapTint, diffuseMapVertexColor -> diffuseTint, diffuseVertexColor.

This is breaking, because you could use old names in procedural material creation code. But I can define old names as get/set too.